src/main/kotlin/com/terraformation/backend/accelerator/db/ActivityMediaStore.kt

```diff
@@ -78,4 +205,29 @@ class ActivityMediaStore(
         .orderBy(ACTIVITY_MEDIA_FILES.FILE_ID)
         .fetch { ActivityMediaModel.of(it) }
   }
+
+  private fun <T> withLockedActivity(activityId: ActivityId, func: () -> T): T {
+    return dslContext.transactionResult { _ ->
+      dslContext
+          .selectOne()
+          .from(ACTIVITIES)
+          .where(ACTIVITIES.ID.eq(activityId))
+          .forUpdate()
+          .fetchOne() ?: throw ActivityNotFoundException(activityId)
+
+      func()
+    }
+  }
+
+  private fun updateListPositions(activityId: ActivityId, fileIds: List<FileId>) {
+    val positionsMap = fileIds.mapIndexed { index, fileId -> fileId to index + 1 }.toMap()
```

Comment:
Can we add a quick javadoc or comment that indicates that `fileIds` is the list of `FileId`s already in the desired order? Initially it seemed like this was just setting the `list_position`s to be in the order of `file_id`s numeric ascending